src/features/common/Layout/Header/accountHeader.tsx

```diff
@@ -0,0 +1,138 @@
+import React,{ useEffect } from 'react';
+import styles from './Header.module.scss';
+import { useRouter } from 'next/router';
+
+import Settings from '../../../../../public/assets/images/icons/userProfileIcons/Settings';
+import SettingsModal from '../../../user/UserProfile/components/SettingsModal';
+import { useAuth0 } from '@auth0/auth0-react';
+import { getRequest, getAccountInfo } from '../../../../utils/apiRequests/api';
+import {
+  setUserExistsInDB,
+  removeUserExistsInDB,
+  getLocalUserInfo,
+  removeLocalUserInfo,
+} from '../../../../utils/auth0/localStorageUtils';
+import BackButton from '../../../../../public/assets/images/icons/BackButton';
+import i18next from '../../../../../i18n'
+const { useTranslation } = i18next;
+
+export default function AcountHeader(props: any) {
+  const [userprofile, setUserprofile] = React.useState();
+  const [forceReload, changeForceReload] = React.useState(false);
+  const { t } = useTranslation(['me']);
+
+  const {
+    isLoading,
+    isAuthenticated,
+    loginWithRedirect,
+    getAccessTokenSilently,
+    logout
+  } = useAuth0();
+  const router = useRouter();
+  const logoutUser = () => {
+    removeLocalUserInfo();
+    logout({ returnTo: `${process.env.NEXTAUTH_URL}/` });
+  }
+
+  React.useEffect(() => {
+    async function loadUserData() {
+      if (typeof Storage !== 'undefined') {
+        let token = null;
+        if (isAuthenticated) {
+          token = await getAccessTokenSilently();
+        }
+        if (!isLoading && token) {
+          try {
+            const res = await getAccountInfo(token)
+            if (res.status === 200) {
+              const resJson = await res.json();
+              setUserprofile(resJson);
+            } else if (res.status === 303) {
+              // if 303 -> user doesn not exist in db
+              setUserExistsInDB(false)
+              if (typeof window !== 'undefined') {
+                router.push('/complete-signup');
+              }
+            } else if (res.status === 401) {
+              // in case of 401 - invalid token: signIn()
+              logoutUser();
+              removeUserExistsInDB()
+              loginWithRedirect({redirectUri:`${process.env.NEXTAUTH_URL}/login`, ui_locales: localStorage.getItem('language') || 'en' });
+            } else {
+              // any other error
+            }
+          } catch (err) {
+            console.log(err);
+          }
+        } else {
+          loginWithRedirect({
+            redirectUri: `${process.env.NEXTAUTH_URL}/account/history`,
+            ui_locales: localStorage.getItem('language') || 'en',
+          });
```

Comment:
What is the difference between specifying the `redirectUri` directly in the `loginWithRedirect` function call or setting them in our localStorage like at https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/f358ad2e551ba81584c090d0ca88c105c59f0b09/pages/manage-projects/add-project/index.tsx#L53-L54?

Reply:
I assume if we want to support a redirectUri for Auth0, we need to add it to all configurations at Auth0 (which could be a lot of work as we already have added quite a lot of applications). Currently `/account/history` is not configured resulting into an error, so we probably need to change this towards using the solution I linked above:

<img width="909" alt="Bildschirmfoto 2021-04-28 um 09 03 48" src="https://user-images.githubusercontent.com/1532418/116360899-afeddd80-a800-11eb-82c3-86fb47094035.png">

Reply:
I saw that someone added http://localhost:3000/account/history to the valid Callback URLs of our development Auth0 tenant `pftp`, but not for the production Auth0 tenant. Thus this will still fail on production. If we want to allow this URL as a callback URL, it has to be done everywhere!

Reply:
There would be at least two production applications on Auth0 to adapt, but this would have to be maintained for every other application (tenant) allowing this URL in the future. I would prefer the code as following:

```
localStorage.setItem('redirectLink','/account/history');
loginWithRedirect({
  redirectUri: `${process.env.NEXTAUTH_URL}/login `,
  ui_locales: localStorage.getItem('language') || 'en',
});
```

Reply:
I change it.